sql/sql_yacc.yy

```diff
@@ -8870,23 +8871,45 @@ opt_for_system_time_clause:
             Item *item= new (thd->mem_root) Item_func_now_local(thd, 6);
             if (item == NULL)
               MYSQL_YYABORT;
-            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_AS_OF, item);
+            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_AS_OF, UNIT_TIMESTAMP, item);
           }
         | FOR_SYSTEM_TIME_SYM
           FROM
           TIMESTAMP simple_expr
           TO_SYM
           TIMESTAMP simple_expr
           {
-            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_FROM_TO, $4, $7);
+            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_FROM_TO, UNIT_TIMESTAMP, $4, $7);
           }
         | FOR_SYSTEM_TIME_SYM
           BETWEEN_SYM
           TIMESTAMP simple_expr
           AND_SYM
           TIMESTAMP simple_expr
           {
-            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_BETWEEN, $4, $7);
+            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_BETWEEN, UNIT_TIMESTAMP, $4, $7);
+          }
+        | FOR_SYSTEM_TIME_SYM
+          AS OF_SYM
+          TRX_ID_SYM simple_expr
```

Comment:
Why do we need it, for trx_id arithmetic, or something like `CONVERT_DATE_TO_TRX_ID()`? 
And if we need it here what I'll get if I put here an `"arbitrary string"`, a bogus result or warning/error?